.github/actions/nexus-associate-tag/action.yml

```diff
@@ -0,0 +1,42 @@
+name: Associate tag with artifacts
+description: Associate tag with artifacts in Nexus
+inputs:
+  tag-name:
+    description: The name of the tag
+    required: true
+  nexus-username:
+    description: Nexus user name
+    required: true
+  nexus-password:
+    description: Nexus password
+    required: true
+  nexus-url:
+    description: Base URL to the Nexus server
+    required: false
+    default: "https://artifacts.alfresco.com/nexus"
+  group-id:
+    description: The maven group-id of the component
+    required: true
+  version:
+    description: The version of the component
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: associate-tag
+      id: associate-tag
+      shell: bash
+      env:
+        TAG_NAME: ${{ inputs.tag-name }}
+        NEXUS_USERNAME: ${{ inputs.nexus-username }}
+        NEXUS_PASSWORD: ${{ inputs.nexus-password }}
+        NEXUS_URL: ${{ inputs.nexus-url }}
+        GROUP-ID: ${{ inputs.group }}
```

Comment:
AFAIK `-` are not allowed in variable names and I think yo can simplify it to `GROUP`